scripts/tests/test_report.py

```
@@ -0,0 +1,134 @@
+from pathlib import Path
+
+
+sys.path.insert(0, str(Path(__file__).resolve().parents[1]))
+
+from localization import build_model, render_html, write_html_report  # noqa: E402
+
+
+class _BalanceChecker(HTMLParser):
+    """Verify that non-void elements are opened and closed in balance."""
+
+    VOID = {"meta", "br", "input", "img", "hr"}
```

Ruff (0.16.0)

[warning] Mutable default value for class attribute

(RUF012)